readthedocs/search/api/v3/serializers.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import serializers
+
+from readthedocs.search.api.v2.serializers import (
+    PageSearchSerializer as PageSearchSerializerBase,
+)
+
+
+class PageSearchSerializer(PageSearchSerializerBase):
+
+    """
+    Serializer for API V3.
+
+    This is very similar to the serializer from V2,
+    with the following changes:
+
+    - ``project`` is an object, not a string.
+    - ``version`` is an object, not a string.
+    - ``project_alias`` isn't present,
+      it is contained in the ``project`` object.
+    """
+
+    project = serializers.SerializerMethodField()
+    version = serializers.SerializerMethodField()
```

Comment:
Why not creating a proper serializer for each of them instead of methods?

Example of this from APIv3 https://github.com/readthedocs/readthedocs.org/blob/8ccd55cf3370e772e8def502fe794386e887daf4/readthedocs/api/v3/serializers.py#L261

Reply:
The only one we can do this for is for the version, for the project we need to call `get_project_alias` to get the alias. The objects these serializers deal with are ES result objects, not RTD objects, we don't have that field in the ES object.